hbase-server/src/main/java/org/apache/hadoop/hbase/ipc/ServerCall.java

```diff
@@ -166,7 +166,7 @@ private void release(int mask) {
   }
 
   @Override
-  public void cleanup() {
+  public synchronized void cleanup() {
```

Comment:
Does this really requires to be _synchronized_? If so, you might also want to synchronize _releaseByWAL_ method?